File: 2_Pregunta/SegundaPregunta.scala
val suma = (x:Int, y:Int) => x+y
val resta = (x:Int, y:Int) => x-y
val multiplicacion = (x:Int, y:Int) => x*y
val division = (x:Int, y:Int) => x/y

def calculadora(x:Int, y:Int, operacion:String):Int={
    |   operacion match {
    |   case "+" => suma(x,y)
    |   case "-" => resta(x,y)
    |   case "*" => multiplicacion(x,y)
    |   case "/" => division(x,y)
    }
}

// se llama 
val operacion = calculadora(2,2,"+")

File: 2_Pregunta/SegundaPregunta.hs
suma :: Fractional a=> a -> a -> a
suma = \ x -> (\ y -> x+y)

resta :: Fractional a=> a -> a -> a
resta = \ x -> (\ y -> x-y)

multiplicacion :: Fractional a=> a -> a -> a
multiplicacion = \x -> (\ y -> x * y)

division :: Fractional a=> a -> a -> a
division = \x -> (\ y -> x / y)

calculadora :: Fractional a => a -> a -> Char -> a
calculadora num1 num2 op
    | op == '+' = suma num1 num2
    | op == '-' = resta num1 num2
    | op == '*' = multiplicacion num1 num2
    | op == '/' = division num1 num2